src/modm/platform/extint/sam/extint.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2020, Erik Henriksson
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include "extint.hpp"
+
+void isr();
+
+namespace modm
+{
+namespace platform
+{
+
+void
+ExternalInterrupt::initialize(ClockGenerator clockGen) {
+	NVIC_DisableIRQ(EIC_IRQn);
+	NVIC_ClearPendingIRQ(EIC_IRQn);
+	NVIC_SetPriority(EIC_IRQn, 0);
+	NVIC_EnableIRQ(EIC_IRQn);
+
+	GenericClockController::enable<Peripheral::Eic>(clockGen);
+
+	// Enable EIC
+	EIC->CTRL.bit.ENABLE = 1;
+	int waitCycles = 2048;
+	while (EIC->STATUS.bit.SYNCBUSY == 1 && --waitCycles);
+}
+
+std::function<void()> ExternalInterrupt::handlers_[16] = {};
```

Comment:
Oh, I did not know that.

Hmm, I am a bit thorn on this one. Yes, of course there will be use cases for closures / member functions and the question is if we should allow that (be an opinionated framework). I think the main motivation for doing any changes here though should be performance / RAM (not penalize everyone because someone wants to do crazy things in ISRs).

I looked through the assembly and we are looking at about 20 instructions for row 44 & 45 which seems like quite a few + there is 16 bytes storage space needed (on ARM 32-bit) due to `sizeof(std::function<void()>)`.

Maybe this is a bit too much overhead for the benefits of std::function, especially since this could be fairly critical code, WDYT?

If we go with function pointers instead, the users can still proxy the call to member function / closure as a workaround.